cypress/integration/data-import/match-on-vendor-reference-number.spec.js

```diff
@@ -0,0 +1,148 @@
+  const itemMappingProfileName = `CaseC350591 Update Item by VRN match ${Helper.getRandomBarcode()}`;
+  const instanceActionProfileName = `CaseC350591 Action for Instance ${Helper.getRandomBarcode()}`;
+  const holdingsActionProfileName = `CaseC350591 Action for Holdings ${Helper.getRandomBarcode()}`;
+  const itemActionProfileName = `CaseC350591 Action for Item ${Helper.getRandomBarcode()}`;
+  const instanceMatchProfileName = `CaseC350591 Match for Instance ${Helper.getRandomBarcode()}`;
+  const holdingsMatchProfileName = `CaseC350591 Match for Holdings ${Helper.getRandomBarcode()}`;
+  const itemMatchProfileName = `CaseC350591 Match for Item ${Helper.getRandomBarcode()}`;
+  const matchProfiles = [
+    {
+      name: instanceMatchProfileName,
+      existingRecordType: 'INSTANCE',
+    },
+    {
+      name: holdingsMatchProfileName,
+      existingRecordType: 'HOLDINGS',
+    },
+    {
+      name: itemMatchProfileName,
+      existingRecordType: 'ITEM',
+    },
+  ];
+  const filePath = 'matchOnVendorReferenceNumber.mrc';
+  const fileName = `vrn${getRandomPostfix()}.mrc`;
+
+  const jobProfilesData = {
+    name: `CaseC350591 Job profile ${Helper.getRandomBarcode()}`,
+    dataType: 'MARC',
+    matches: [
+      {
+        matchName: instanceMatchProfileName,
+        actionName: instanceActionProfileName,
+      },
+      {
+        matchName: holdingsMatchProfileName,
+        actionName: holdingsActionProfileName,
+      },
+      {
+        matchName: itemMatchProfileName,
+        actionName: itemActionProfileName,
+      },
+    ]
+  };
+
+  before(() => {
+    cy.createTempUser([
+      permissions.uiOrdersView.gui,
+      permissions.uiOrdersCreate.gui,
+      permissions.uiOrdersEdit.gui,
+      permissions.uiOrdersDelete.gui,
+      permissions.inventoryAll.gui,
+      permissions.moduleDataImportEnabled.gui,
+      permissions.settingsDataImportEnabled.gui,
+      permissions.dataImportDeleteLogs.gui,
+      permissions.uiQuickMarcQuickMarcBibliographicEditorView.gui,
+    ]).then(userProperties => {
+      userId = userProperties.userId;
+      cy.login(userProperties.username, userProperties.password);
+    });
+
+    cy.readFile(`cypress/fixtures/${filePath}`).then(content => {
+      FileManager.createFile(`cypress/fixtures/${fileName}`, content);
+    });
+  });
+
+  after(() => {
+    users.deleteViaApi(userId);
+  });
+
+  it('C350591 Match on VRN and update related Instance, Holdings, Item', { tags: [TestTypes.smoke] }, () => {
+    // create order
+    cy.visit(TopMenu.ordersPath);
+    Orders.createOrder(order);
+    MatchOnVRN.verifyCreatedOrder(order);
+
+    // add and verify po line
+    Orders.createPOLineViaActions();
+    MatchOnVRN.fillPOLineInfo();
+    MatchOnVRN.goBackToPO();
+    MatchOnVRN.openOrder();
+    MatchOnVRN.verifyOrderStatus();
+
+    // create field mapping profiles
+    cy.visit(SettingsMenu.mappingProfilePath);
+    MatchOnVRN.creatMappingProfilesForInstance(instanceMappingProfileName)
```

Comment:
step in TestCase and implementation should be fully crossed. Think to redesign related steps or update your implementation

Reply:
could you clarify what step is not implemented? I checked test case and my implementation has the same steps as in the testcase